apps/src/dance/lab2/views/DanceView.tsx

```diff
@@ -1,19 +1,36 @@
 import Instructions from '@cdo/apps/lab2/views/components/Instructions';
 import PanelContainer from '@cdo/apps/lab2/views/components/PanelContainer';
-import React from 'react';
+import AgeDialog, {getFilterStatus} from '@cdo/apps/templates/AgeDialog';
+import {CurrentUserState} from '@cdo/apps/templates/CurrentUserState';
+import React, {useCallback, useState} from 'react';
+import {TypedUseSelectorHook, useSelector} from 'react-redux';
 import moduleStyles from './dance-view.module.scss';
 const commonI18n = require('@cdo/locale');
 
 const DANCE_VISUALIZATION_ID = 'dance-visualization';
 const BLOCKLY_DIV_ID = 'dance-blockly-div';
 
+const useTypedSelector: TypedUseSelectorHook<{
```

Comment:
Could you explain why you defined `useTypedSelector` in this way? Is it just to simplify syntax for lines 22,23?

Reply:
Yeah, I just happened to come across this recently ([docs](https://react-redux.js.org/using-react-redux/usage-with-typescript#define-typed-hooks)). Since our root store isn't typed, we've previously always manually typed the slice of the state we're dealing with when using the useSelector hook (ex. something like `useSelector((state: {lab: LabState}) => state.lab.xyz)`. The `TypedUseSelectorHook` just allows us to define all the typing we need in one go. For now we're only referencing the currentUserState, but this would probably expand to include other slices of state that we're referencing in the component, and then we we actually call `useSelector`, we don't have manually assign a type within the function.

Reply:
Awesome - thanks for the explanation!